libarchive/archive_read_support_format_tar.c

```diff
@@ -2156,10 +2158,21 @@ pax_attribute_LIBARCHIVE_xattr(struct archive_entry *entry,
 	/* Base-64 decode value */
 	value_decoded = base64_decode(value, value_length, &value_len);
 	if (value_decoded == NULL) {
+		archive_set_error(&a->archive, ARCHIVE_ERRNO_MISC,
+		    "Failed to decode xattr `%.*s'", (int)name_length, name);
 		free(name_decoded);
 		return 1;
 	}
 
+	if (value_len > xattr_limit) {
```

Comment:
This is the wrong place to check the limit.   By removing the other check and checking here, you're still dropping overlarge xattrs, but now you're needlessly allocating memory for them.  It's better to check the size before trying to read it and skipping it.